views/js/runner/plugins/tools/textToSpeech/textToSpeech.js

```diff
@@ -0,0 +1,374 @@
+                    self.trigger('stop');
+                });
+            },
+
+            /**
+             * Re-initialize texthelp
+             */
+            updateTexthelpCache: function updateTexthelpCache(deliveryId, itemId) {
+                var tss = this._get('texthelpSpeechStream');
+
+                tss.g_strBookId = deliveryId;
+                tss.g_strPageId = itemId;
+
+                this._exec('tagSentences', options.$contentArea.selector);
+            },
+
+            /**
+             * Enable texthelp
+             */
+            enable: function enable() {
+                this._exec('enableSpeech');
+            },
+
+            /**
+             * Disable texthelp
+             */
+            disable: function disable() {
+                this._exec('disableSpeech');
+            },
+
+            /**
+             * Play
+             */
+            play: function play() {
+                if (!this._exec('isSpeaking') || this._exec('isPaused')) {
+                    this._exec('event_play');
+                    this.trigger('play');
+                }
+            },
+
+            /**
+             * Pause
+             */
+            pause: function pause() {
+                this._exec('event_pause');
+                this.trigger('pause');
+            },
+
+            /**
+             * Stop
+             */
+            stop: function stop() {
+                this._exec('stopSpeech');
+                this.trigger('stop');
+            },
+
+            /**
+             * Speech speed
+             */
+            setSpeed: function setSpeed(e, value) {
+                speed = +value;
+
+                this._exec('setSpeedValue', speed);
+                this.trigger('setSpeed', speed);
+            },
+
+            /**
+             * Volume
+             */
+            setVolume: function setVolume(e, value) {
+                volume = +value;
+
+                this._exec('setVolumeValue', volume);
+                this.trigger('setVolume', volume);
+            },
+
+            /**
+             * Click to pronounce
+             */
+            clickToSpeak: function clickToSpeak() {
+                options.enableClickToSpeak = !options.enableClickToSpeak;
+                this._exec('enableClickToSpeak', options.enableClickToSpeak);
+                this._exec('enableContinuousReading', !options.enableClickToSpeak); // continuous reading is off when click to speak is on
+                this.trigger('clickToSpeak');
+            }
+        }, {
+            // defaults
+        })
+        .setTemplate(tpl)
+        .on('init', function () {
+            var self = this;
+
+            speed = 40; // default speed
+            volume = 40; // default volume
+
+            // we have to mark some blocks as ignored to prevent TTS accessing it
+            $(options.ignoreEls.join(',')).each(function () {
+                $(this).attr('ignore', true);
```

Comment:
Would it not be simpler to write `$(options.ignoreEls.join(',')).attr('ignore', true);`?

Reply:
Yes, good call.